crates/q_cli/src/cli/chat/tools/mod.rs

```
@@ -194,6 +195,14 @@ impl ToolPermissions {
         self.permissions.clear();
     }
 
+    pub fn reset_tool(&mut self, tool_name: &str) {
+        if !self.permissions.contains_key(tool_name) {
+            warn!("No custom permissions set for tool '{tool_name}' to reset");
```

Will this print to the user? IIRC it does not. Instead, you may want to throw an error from this function and catch it when handling `ToolsSubcommand::ResetSingle`. See other comment